subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -774,23 +775,24 @@ void pe_soft_reset_run(void *obj)
 	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_SEND_SOFT_RESET)) {
 		/* Send Accept message */
 		pe_send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_ACCEPT);
-		return;
 	}
-
 	/*
 	 * The Policy Engine Shall transition to the PE_SNK_Wait_for_Capabilities
 	 * state when:
 	 *	1: The Accept Message has been sent on SOP.
 	 */
-	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+	else if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
 		pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
 	}
 	/*
 	 * The Policy Engine Shall transition to the PE_SNK_Hard_Reset
 	 * state when:
 	 *	1: The Protocol Layer indicates that a transmission error
```

Comment:
The spec also mentions SenderResponseTimer here.

Reply:
I have another PR that creates a Sender Response Timer Super state that fixes this issue.